python/pyiceberg/io/pyarrow.py

```diff
@@ -358,14 +368,17 @@ def field(self, field: NestedField, field_result: pa.DataType) -> pa.Field:
             name=field.name,
             type=field_result,
             nullable=field.optional,
```

Comment:
@Fokko, looks like there's an issue with pyarrow here. Python gives a warning that this is not an argument to the function (which has `bool_nullable`) but that's probably not correct because the Python is a stub. Might be something we can fix upstream in pyarrow.

Reply:
Missed this one. I looked into this earlier, but it isn't really trivial to fix this. It is part of the Cython generator that generates docs, which are then interpreted by pycharm.